src/hooks/Trades.ts

```diff
@@ -4,12 +4,17 @@ import flatMap from 'lodash.flatmap'
 
 import { useActiveWeb3React } from './index'
 import { usePairs } from '../data/Reserves'
+
 import { BASES_TO_CHECK_TRADES_AGAINST } from '../constants'
 
 function useAllCommonPairs(tokenA?: Token, tokenB?: Token): Pair[] {
   const { chainId } = useActiveWeb3React()
 
   const bases = useMemo(() => BASES_TO_CHECK_TRADES_AGAINST[chainId as ChainId] ?? [], [chainId])
+  const commonBases = useMemo(
+    () => flatMap(bases, (base): [Token, Token][] => bases.map(otherBase => [base, otherBase])),
+    [bases]
+  )
 
   const allPairs = usePairs([
```

Comment:
sorry i meant this list of pairs which is the combination of all the other lists, if you memoize the whole list you probably don't have to memoize `commonBases`